### DISCUSSION TITLE
Advice Needed on Managing Simulation Stability During Sudden High-Intensity Events like Fusion

### DISCUSSION THREAD

**Discussion by NickHaught02** [BLAST-WarpX/warpx#5018] [Q&A] [answered]
Hello,
I am running high-intensity plasma simulations, specifically focusing on scenarios where fusion events occur within the simulation. I have encountered issues where the simulation becomes unstable or crashes, particularly during these high-energy fusion events. I  can solve this by using more GPUs and of course, lowering the resolution. However, I wanted to see how efficient I could make my simulations while using a higher resolution. I have tried additional settings like load balancing which have helped a little, but its just how quickly the fusion events happen that crashes the sims. 

What suggestions are there on optimizing load balancing or particle handling during sudden increases in particle interactions and energy densities? Are there any specific features (other than adaptive mesh) that could help manage more effectively?

Thank you for any suggestions,
Nick

**Comment by ax3l** [comment, marked as answer]
Thanks for posting, @NickHaught02!

>  I have encountered issues where the simulation becomes unstable or crashes, [...] I can solve this by using more GPUs and of course, lowering the resolution.

You mean the simulation _crashes because it runs out of memory_, because you create more particles dynamically?

> I have tried additional settings like load balancing which have helped a little, but its just how quickly the fusion events happen that crashes the sims.

That is definitely the right way to go. Additionally, the fusion algorithms right now assume that the events are quite rare compared to the overall plasma dynamics. To counter the rapid increase of numerical particles, you can try resampling/merging the particles via `<species>.do_resampling`, especially the `velocity_coincidence_thinning` algorithm.
https://warpx.readthedocs.io/en/latest/usage/parameters.html#particle-initialization

**Reply by NickHaught** [reply]
Thank you for the suggestion!
